### ISSUE TITLE
Bug in getting started doc page

### ISSUE THREAD

**Issue by y030** [mosra/magnum#224]
Hey!

Sorry but I couldn't locate the getting started doc page in any repository (I haven't looked exhaustively, though). 

I believe 
```
Debug() << "Hello! This application is running on" << Context::current().version()
              << "using" << Context::current().rendererString();`
```
should be 
```
Debug() << "Hello! This application is running on" << Context::current().versionString()
              << "using" << Context::current().rendererString();
```

**Comment by mosra** [comment]
Hi!

Hmm. When you use `version()`, it prints `Invalid` instead of a proper version, right? That's because Magnum doesn't really know about GL 4.6 yet. Let me fix that. Other than that, the `version()` is correct.

**Comment by y030** [comment]
As a matter of fact, that isn't why.
My code won't compile if I leave version().

I am getting the following error
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2679	binary '<<': no operator found which takes a right-hand operand of type 'const Magnum::Version' (or there is no acceptable conversion)	KVRS	C:\develop\dependencies\include\Corrade\Utility\Debug.h	923	
````

**Comment by mosra** [comment]
Ah. Then it's just a missing include:

```cpp
#include <Magnum/Version.h>
```

**Comment by y030** [comment]
You are right. I am having some (to me at least) strange behavior.
I am using the qt base bootstrap project.

Header
```

/* Magnum GL headers must always be included before Qt ones */
#include <Magnum/DefaultFramebuffer.h>
#include <Magnum/Platform/Context.h>

#include <Magnum/Version.h>
#include <Magnum/Math/Color.h>

#include <QtWidgets/QApplication>
#include <QtWidgets/QOpenGLWidget>

#include <optional>
```

and cpp
```
#include "header"
...

void KVRS::initializeGL() {
  _context.emplace(Platform::Context{0, nullptr});

  /* TODO: Add your initialization code here */

  Debug() << "This application is running on" << Context::current().version()
    << "using" << Context::current().rendererString();
}

void KVRS::paintGL() {
  defaultFramebuffer.clearColor(Math::Color4<Magnum::Float>(1.0f, 0.0f, 0.0f));

  /* TODO: Add your drawing code here */
}
...
```
does not work. Both for `version()` and also `Math::Color4<Magnum:Float>` ...
It does work though when I move the following includes from the header to the cpp.
```
#include <Magnum/Version.h>
#include <Magnum/Math/Color.h>
```

**Comment by mosra** [comment]
Huh, this is strange, seems like there is some symbol conflict with Qt or Windows headers :/ I can't reproduce anything like that on my Linux box.

What's the exact compiler error?

**Comment by y030** [comment]
If I leave both lines uncommented I get the following output
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2027	use of undefined type 'Magnum::Math::Color4<Magnum::Float>'	
KVRS	C:\develop\KVRS\src\KVRS.cpp	18	
```

If I comment out the related line `defaultFramebuffer.clearColor(Math::Color4<Float>(0.0f, 0.0f, 0.0f));`
I get the following error
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2679	binary '<<': no operator found which takes a right-hand operand of type 'const Magnum::Version' (or there is no acceptable conversion)	
KVRS	C:\develop\dependencies\include\Corrade\Utility\Debug.h	923	
```

**Comment by mosra** [comment]
This is *really* weird. Are you sure that the header actually gets included (and not some other instead)? Are you sure that the `#include`s are not accidentally wrapped in some namespace declaration or such?

Can you post the complete code to https://gist.github.com, for example?

**Comment by y030** [comment]
I find this really weird too, thank you for your help!

Here is the gist: https://gist.github.com/y030/a335857359cf07053709c11fe8de8826

**Comment by mosra** [comment]
Nope, I don't see anything that would cause the above problem.

The only possible cause I see is that there's some other 'KVRS.h` that gets included instead of this one. Try to rename the `KVRS.h` header to something else to really rule that out.

Otherwise, I don't know. Sorry. Everything compiles fine here.

**Comment by y030** [comment]
I deleted the whole folder, regenerated it with cmake and set everything up as it was.
This fixed it. I believe I might have broken something in the old CMakeLists.txt or in the project properties inside VS2017 in my first try.

Thank you very much for your help nevertheless. Turns out the code is totally fine!

**Comment by mosra** [comment]
Great.

I will still keep this open until I fix the `Invalid` error mentioned above :)

**Comment by mosra** [comment]
Minimal OpenGL 4.6 support implemented in 20a499aa0f6735520365350b77fc98bd4af54375, so closing this :)
